ethcore/src/tx_filter.rs

```diff
@@ -83,24 +83,26 @@ impl TransactionFilter {
 		let contract_address = self.contract_address;
 
 		// Check permissions in smart contracts
-		let permissions = self.contract.functions()
+		let (permissions, filter_only_sender) = self.contract.functions()
```

Comment:
I think we should keep caching for performance reasons, so I came up with an idea to have 2nd return value in the contract function, which will define whether filtering depends on to address and value in ether and if not, sender address will be cached which will increase performance.